docarray/array/mixins/match.py

```diff
@@ -70,6 +71,11 @@ def match(
         for d in self:
             d.matches.clear()
 
+        if on is not None:
+            find_kwargs = {'on': on}
```

Comment:
why {'on': on} will not directly work regardless of it being None?

Reply:
This is a relic from when not all backends had an `on` parameter. Will remove.